applications/minotari_app_grpc/src/authentication/basic_auth.rs

```diff
@@ -77,17 +92,40 @@ impl BasicAuthCredentials {
         Ok(credentials)
     }
 
+    // This function provides a constant time comparison of the given username with the registered username.
+    fn constant_time_compare_username(&self, username: &str) -> bool {
+        let a_bytes = self.user_name.as_bytes();
+        let b_bytes = username.as_bytes();
+
+        let valid_username = !(b_bytes.is_empty() || b_bytes.len() > MAX_USERNAME_LEN);
```

Comment:
It's safe to return early if the supplied username is malformed, since doing so doesn't leak anything about the validity of the supplied username.

Reply:
Doing so will also let you avoid having to use any clipping.

Reply:
This is intended behaviour by design; I would like any result given by this function to perform in constant time.